challenges/backend/src/app/services/CarOnSaleClient/interface/ICarOnSaleClient.ts

```diff
@@ -1,8 +1,13 @@
+import { Auction } from "../dtos/auction/auction";
+import { AuthenticationResponse } from "../dtos/auth/AuthenticationResponse";
+import { IActionFilter } from "./IActionFilter";
+
 /**
  * This service describes an interface to access auction data from the CarOnSale API.
  */
 export interface ICarOnSaleClient {
 
-    getRunningAuctions(): Promise<any /* TODO: Introduce a type */>
-
+    getRunningAuctions(token: string, filter: IActionFilter, count: boolean): Promise<Auction>
+    userAuthentication(): Promise<AuthenticationResponse>
+    print(): Promise<void>
```

Comment:
As per the interface docs, this class/interface should be responsible to interact with the CarOnSale API. Processing the results does not seem to be perfectly fitting this purpose. I would expect this to be in another class, responsible for calling the client and processing results. Could be in the Calculator, maybe? Or perhaps in another layer?

Additionally, the method naming is not super descriptive.